### ISSUE TITLE
Using ref Property

### ISSUE THREAD

**Issue by zaneevat** [uderline/openapi-php-attributes#16]
How to describe nested objects in the request body?
```
{
    "service": "service key",
    "shippingAddress": {
        "city": "city",
        "isTerminal": false
    },
    "deliveryAddress": {
        "city": "city",
        "address": "address",
        "isTerminal": false
    }
}
```
I tried to implement the following structure:
```
    #[
        POST('/delivery/calculation', [], 'Delivery Calculation'),
        Property(Type::REF, CalculationRequest::class),
        Response(200, ref: CalculationResponse::class)
    ]
    public function calculation(CalculationRequest $request): JsonResponse
    {
    }
```
```
#[
    Schema(name: CalculationRequest::class),
    Property(PropertyType::STRING, "service"),
    Property(PropertyType::OBJECT, "shippingAddress"),
    Property(PropertyType::OBJECT, "deliveryAddress"),
]
class CalculationRequest extends JsonRequest implements ValidateRequestInterface
{
    private string $service;
    private Address $shippingAddress;
    private Address $deliveryAddress;
}
```

I couldn't find anything similar in the documentation. There is an OpenApiGenerator\Attributes\RefProperty class in the source code, but I couldn't use it, maybe it's for other purposes.

**Comment by uderline** [comment]
Hi !

`RefProperty` is only used for responses. I had something in mind like:
```
#[Property(["field1.subfield1" => "string", "field1.subfield2" => "int])]
```

But now I'm thinking of automatically generating the RequestBody ref directly from the Request object (in your case `CalculationRequest`). Basically this means you would declare your schema in your request file and in your controller, just the POST and Response.

Maybe this would include implementing an interface to "find" the request body or just any class inheriting symphony's Request object.

What do you think ?

_Obviously I'm open to PRs but if needed I can do it_

**Comment by zaneevat** [comment]
The variant with the Request object looks very interesting. But in general, this will not solve my problem, it would be great to explicitly specify child schemas. In my example, it would look something like this:

```
#[
    Schema(name: CalculationRequest::class),
    Property(PropertyType::STRING, "service"),
    Property(PropertyType::REF, "shippingAddress"),
    PropertyRef(Address::class),
    Property(PropertyType::REF, "deliveryAddress"),
    PropertyRef(Address::class),
]
class CalculationRequest extends JsonRequest implements ValidateRequestInterface
{
    private string $service;
    private Address $shippingAddress;
    private Address $deliveryAddress;
}
```
```
#[
    Schema(name: Address::class),
    Property(PropertyType::STRING, "region"),
    Property(PropertyType::STRING, "city"),
    Property(PropertyType::STRING, "address"),
    Property(PropertyType::STRING, "fullAddress"),
    Property(PropertyType::BOOLEAN, "isTerminal"),
]
class Address
{
    public string $region;
    public string $city;
    public string $address;
    public string $fullAddress;
    public bool $isTerminal;
}
```
How do you think?

I think I can implement this in a separate PR.

**Comment by uderline** [comment]
That looks good !

Maybe the syntax should be like the `Response` attribute where you can put a reference in the attributes.

You could then have:
```php
#[
    Schema,
    Property(name: "shippingAddress", ref: Address::class)
]
class CalculationRequest {
}
```

Though, this could just be a shortcut ...

**Comment by uderline** [comment]
Published in version 1.2 and 1.3

**Comment by zaneevat** [comment]
Thank you @uderline 🙏
